### DISCUSSION TITLE
[Open Planning R24.08] Brainstorming Features

### DISCUSSION THREAD

**Discussion by stefan-ettl** [eclipse-tractusx/tractusx-edc#1162] [General]
This Discussion should be a placeholder to create a Brainstorming List of Features to be addressed in the 24.08 Release for the TractusX-EDC.

In order to contribute your idea or request, please copy the following table and add your content accordingly.
Please take care of a consecutive number in regards to your predecessor post.

| Number    | Title  | Description  |
| -------- | ------- | ------- |
| 1  | Concept for a Connector Testbed | First Concept for a Connector Test Kit with a Catena-X Profile (testing DSP & IATP) |
| 2 | MXD - Wallet Upgrade | Upgrade the MXD Framework to support a new MIW Version that follows the IATP (Identity and Trust Protocol) |
| 3 | Archiving of Metadata | Follow up  |

**Comment by lgblaumeiser** [comment]
Based on https://github.com/eclipse-tractusx/tractusx-edc/discussions/1087

| Number  | Title | Description |
| ------------- | ------------- | ------------- |
| 3 | Delegate api-key management to an external IdP  | Delegate the responsibility of an IdP by managing the api-key of the management api internally to an external IdP, e.g., Keycloak  |
| 4 | Allow authentication to the mangement api with OAuth  | In addition of using an api-key, allow usage oauth for authentication of requestors in the mangement api  |
| 15 | Architecture Documentation  | Create a structural overview of the EDC that describes the components and general mechanisms like the extension mechanism |

**Comment by jimmarino** [comment]
| Number  | Title | Description |
| ------------- | ------------- | ------------- |
| 5 | Management Domains  | *Upstream EDC* Support distributed deployment scenarios where an organization may have different departments that manage specific sets of assets for data sharing.  |
| 6 | Transfer formats| *Upstream EDC* Support different transfer format combinations that influence data plane selection. For example, a data plane may support only specific transfer formats. This is a significant amount of work.|
| 7 | Bi-directional transfers |  *Upstream EDC* Some data sharing scenarios require both the provider and consumer to supply transfer metadata. Support the ability to exchange transfer metadata in a full-duplex fashion. | 
| 8 | DSP and IATP fixes | With the DSP and IATP standardization effort starting, we will need to allocate time to tracking spec changes, bug fixing, and ensuring interoperability.|
| 9 | CatalogCrawler Distribution | Many end-users and downstream projects are unaware of the EDC Catalog Crawler or how to execute catalog queries. Create a TX distribution for the Catalog Crawler to facilitate its use. |
| 10 | Product Hardening | We need to devote effort to runtime hardening and bug fixing. |

**Comment by paullatzelsperger** [comment]
| Number  | Title | Description |
| ------------- | ------------- | ------------- |
| 11 | Extension hygiene  | Getting rid of unsupported/not officially sanctioned extensions  |
| 12 | Extension refactoring | Rewrite/redesign legacy extensions that aren't up to Tx-EDC standards and coding guide lines |
| 13 | Upgrade VC Suite | The library we use for handling LDP-VCs had a major upgrade with lots of breaking changes. We either should adapt the code base or roll our own. https://github.com/eclipse-edc/Connector/issues/4066 |
| 14 | Settings hygiene | All Tractus-X EDC properties should start with `tx.edc.` to be able to clearly distinguish them. We should go through all extensions and fix that. |

**Comment by rafaelmag110** [comment]
| Number  | Title | Description
| ------------- | ------------- | ------------- |
| 16  |  Rethink secret storing for DMA users | _Upstream EDC_ The users of the DMA can define secrets aliases in data addresses and data destinations to later be resolved by the EDC. In EDC deployments that integrate with hashicorp vault, users of the EDC can also have access to the vault instance and store their secrets by themselves. However, this is not the case for cloud stores such as AWS Secrets Manager or Azure Vault. There could be a way to enable users to store their secrets in such vaults. |

**Comment by stefan-ettl** [comment]
PI13 Brainstorming phase closed - Issues created.
